### DISCUSSION TITLE
Modern Search can´t correct filter

### DISCUSSION THREAD

**Discussion by gitulm** [microsoft-search/pnp-modern-search#1876] [Q&A] [answered]
In our UPS we have a field "RefinableString130" where users can add multiple entries e.g. HAT,HADW. As soon as two values are present, the search finds nothing.

Actual behavior: 

"data": { 

    "totalItemsCount": 0, 
    "items": [], 
    "filters": [], 
    "queryModification": "(RefinableString130:HAT|HADW) -ContentClass=urn:content-class:SPSPeople", 
    "secondaryResults": [] 
  } 

Expected behavior (debugging template) 

"data": { 
    "totalItemsCount": 0, 
    "items": [], 
    "filters": [], 
    "queryModification": "(RefinableString130:KPS OR RefinableString130:KWH) -ContentClass=urn:content-class:SPSPeople", 
    "secondaryResults": [] 
  } 
 
We used versions 4.5.3 and 4.5.4 and both got the same result. Is this a bug in the PnP Modern Search code?

**Comment by wobba** [comment, marked as answer]
@gitulm I assume the value of `RefinableString130` value is `HAT|HADW`. AS the web parts are not aware of how values are stored in your properties we cannot automatically just split them on `|` and assume that's the right thing.

It would also help if you shared a bit about your setup and where/how the query is being constructed with your value.

**Reply by gitulm** [reply]
I'm trying to explain our setup and I hope this is the right answer. In the UPS we have an attribute c-einrichtung (which contains taxonomy values), which can contain one or more values. E.g. HAT or HAT|HADW|HAS. The c-einrichtung attribute is linked to the managed property Refinablestring130.
Our search query is as follows:
Path:https://xxxx.sharepoint.com/
{|RefinableString130:{User.C setup}}

If there is only one value, e.g. HAT, then the corresponding documents are displayed. If there are multiple HAT|HADW values, nothing is displayed.

Did I make myself clear?

**Reply by wobba** [reply]
Turns out it was a bug. Fixed now and will be in the next v4 release. #1879
